src/components/crossDevice/MobileFlow.js

```diff
@@ -0,0 +1,38 @@
+import { h, Component } from 'preact'
+
+import MobileConnected from './MobileConnected'
+import MobileComplete from './MobileComplete'
+
+class MobileFlow extends Component {
+  componentDidMount() {
+    this.props.socket.on('get config', this.sendConfig)
+    this.props.socket.on('complete', this.onMobileComplete)
+  }
+
+  componentWillUnmount() {
+    this.props.socket.off('get config')
+    this.props.socket.off('complete')
+  }
+
+  sendConfig = (data) => {
+    if (this.props.roomId && this.props.roomId !== data.roomId) {
```

Comment:
I prefer the style of declaring in the first line all the props which are used in the method. That way you can quickly skim for prop dependencies, without having to read the fully method body